### DISCUSSION TITLE
How to mark data from multiple clusters

### DISCUSSION THREAD

**Discussion by EsDmitrii** [grafana/mimir#5725] [Help and support] [answered]
Hi!
I'm new to Mimir and I can't find any related docs about how to mark data from multiple clusters using ruler (may be it's not ruler's work).
I set up Mimir using [these](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/small.yaml) values and official [docs](https://grafana.com/docs/mimir/latest/get-started/).
I use s3 minio to store data and this is how it looks now:
<img width="293" alt="Screenshot 2023-08-14 at 11 26 31" src="https://github.com/grafana/mimir/assets/124890665/8a6e9b56-fa1d-49aa-8da8-f3e202524d7e">
As you can see, there is no structuring on any basis, folder names have incomprehensible names, and etc.

I created two VictoriaMetrics Agents in two clusters that push data to Mimir. 
I want to mark data or create its own folders for every cluster something like cluster name == folder name where data is stored.
What should I use? I think I should configure ruler, but I can't find any documentation how to do it using helm chat.
Can I write all data ruler, tsdb, alermanager, etc to one s3 or do I need to create multiple buckets for these components?

And another thing that I found, that Mimir use user `anonymous` to push data (why?). How can I change the name of this user or change the folder name 
<img width="293" alt="260397220-8a6e9b56-fa1d-49aa-8da8-f3e202524d7e" src="https://github.com/grafana/mimir/assets/124890665/de4a7a26-ed2d-482e-a956-868a081bd8b7">

May be someone has reference values to deploy production-ready Mimir for multiple k8s clusters? (in my case there are a lot of k8s, about 50 clusters)

Thank you in advance!

**Comment by pstibrany** [comment]
Hello,

> I want to mark data or create its own folders for every cluster something like cluster name == folder name where data is stored.
> What should I use? I think I should configure ruler, but I can't find any documentation how to do it using helm chat.

`anonymous` is simply a tenant name. You need to

1) enable multitenancy in Mimir
2) set up your vmagent (or Grafana Agent, or Prometheus, or anything that writes data to Mimir) to send `X-Scope-OrgID` HTTP header, when sending data to Mimir. Value of that header will be "tenant", Mimir will store data for different tenants into separate parts of the storage. (Looks like vmagent supports that feature: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2805).

(Each tenant in Mimir can have separate set of alerting rules or configuration options)

**Reply by EsDmitrii** [reply]
Hi!
Thank you!
Will try.

Now I turned on tests (set remote write to Mimir from little prod cluster) and I faced a lot of 400/429 errors
As I understand there no deduplication enabled by default? 
Cant find any related docs for helm.
I only found this https://grafana.com/docs/mimir/latest/operators-guide/mimir-runbooks/#err-mimir-tenant-max-ingestion-rate, but how to set `-distributor.ingestion-rate-limit` via Helm?

This agent writes ~15K samples per second. This is little prod cluster, not huge.. and this is only the beginning

**Reply by EsDmitrii** [reply]
thanks to the Medium that it is exists:)
Found this topic https://tech.loveholidays.com/grafana-mimir-our-journey-towards-infinite-wisdom-with-5m-active-time-series-7a262ba53a3f

Where can I find all values and options that I can set?

**Reply by pstibrany** [reply, marked as answer]
> Where can I find all values and options that I can set?

You can check values.yaml file in Helm chart https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/values.yaml, and for complete list of command line and YAML options you can consult https://grafana.com/docs/mimir/latest/references/configuration-parameters/

**Reply by EsDmitrii** [reply]
thank you a lot
this is what I need
